x-pack/plugins/monitoring/public/views/base_controller.js

```diff
@@ -129,11 +153,21 @@ export class MonitoringViewBaseController {
     // needed for chart pages
     this.onBrush = ({ xaxis }) => {
       const { to, from } = xaxis;
+      const fromTime = moment(from);
+      let toTime = moment(to);
+      if (toTime.unix() - fromTime.unix() < 40) {
+        toTime = fromTime.clone().add(40, 'seconds');
+      }
+
+      removePopstateHandler();
       timefilter.setTime({
-        from: moment(from),
-        to: moment(to),
+        from: fromTime,
+        to: toTime,
         mode: 'absolute'
       });
+      ++zoomInLevel;
+      clearTimeout(deferTimer);
+      setTimeout(() => addPopstateHandler(), 10);
```

Comment:
Is `10` significant here too?

Reply:
Triggering a back event and then adding the change state event triggers it immediately. To avoid this I'm differing it 10ms, although, it could technically be 0ms (since I just want to do this on the next code digest)

Reply:
Hmm. I'm not sure I'm seeing the same behavior. I tried removing the `setTimeout` and the functionality seems the same. What should I be seeing instead?

Reply:
Without timeout `popstateHandler` will get triggered by `popstate`, thus changing `zoomInLevel` back to zero. So, you'll never actually get to see the Zoom Out button